tools/swift-demangle-fuzzer/swift-demangle-fuzzer.cpp

```diff
@@ -0,0 +1,29 @@
+//===--- swift-demangle-fuzzer.cpp - Swift fuzzer -------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This program tries to fuzz the demangler shipped as part of the swift
+// compiler.
```

Comment:
I would add instructions to this comment saying:

1. For this to work you need to pass --enable-sanitizer-coverage to build-script since otherwise (insert explanation here).
2. I would add as well that for best results this should be used with sanitized swift host tools and then mention asan/lsan/etc.